database/bigquery.js

```diff
@@ -0,0 +1,87 @@
+const fs = require('fs');
+const EventEmitter = require("events");
+const {BigQuery} = require('@google-cloud/bigquery');
+
+const isProduction = process.env.NODE_ENV && process.env.NODE_ENV === 'production';
+
+class RecordBuffer extends EventEmitter {
+    constructor(interval = 5 * 60 * 1000, bufferSize = 100) {
+        super();
+
+        this.interval = interval;
```

Comment:
this looks like we'll use `setInterval`, perhaps `bufferTimeout` would fit better the intention

Reply:
but its periodic in nature - the reason for setTimeout is that we are going to interrupt and restart the timer on every flush

maximumTimeBetweenWrites?

Reply:
yep I think `setTimeout` is the right approach, just that by reading the code `interval` reminded me of `setInterval`, which might lead to confusion, minor issue anyway 😊 

Your name suggestion is a bit long but indeed meaningful!